applications/ShallowWaterApplication/custom_utilities/interpolate_sw_to_pfem_utility.hpp

```diff
@@ -0,0 +1,174 @@
+
+    void InterpolateVariables (ModelPart& rPfemModelPart, ModelPart& rSWModelPart)
+    {
+        
+        const int NNodesSW = static_cast<int>(rSWModelPart.Nodes().size());
+        ModelPart::NodesContainerType::iterator node_beginSW = rSWModelPart.NodesBegin();
+        
+        const int NNodesPfem = static_cast<int>(rPfemModelPart.Nodes().size());
+        ModelPart::NodesContainerType::iterator node_beginPfem = rPfemModelPart.NodesBegin();
+
+        double distance, distance1, distance2;
+        double ratio, ratio1,ratio2;
+        double tol = 1e-4;
+                
+        int     j1, j2;
+        
+        double             height, vertical_velocity, topography;
+        array_1d<double,3> velocity;
+
+        for(int i = 0; i < NNodesPfem; i++) {
+            ModelPart::NodesContainerType::iterator it_node = node_beginPfem + i;
+
+            distance1 = 1e32;
+            distance2 = 1e32;
+
+            for (int j = 0; j < NNodesSW; j++){
+            ModelPart::NodesContainerType::iterator jt_node = node_beginSW + j;
+
+                distance = (it_node->X() - jt_node->X())*(it_node->X() - jt_node->X()) + (it_node->Y() - jt_node->Y())*(it_node->Y() - jt_node->Y());
+                if (distance < distance1){
+                       distance1 = distance;
+                       j1 =  j;
+                }
+            }
+
+            for (int j = 0; j < NNodesSW; j++){
+            ModelPart::NodesContainerType::iterator jt_node = node_beginSW + j;
+
+                distance = (it_node->X() - jt_node->X())*(it_node->X() - jt_node->X()) + (it_node->Y() - jt_node->Y())*(it_node->Y() - jt_node->Y());
+                if ((distance < distance2) & (distance > distance1)){
+                       distance2 = distance;
+                       j2 =  j;
+                }
+            }
+            
+            ModelPart::NodesContainerType::iterator jt_node1 = node_beginSW + j1;
+            ModelPart::NodesContainerType::iterator jt_node2 = node_beginSW + j2;
+
+            double dx = fabs(jt_node1->X() - jt_node2->X());
+            double dy = fabs(jt_node1->Y() - jt_node2->Y());
+            
+            if (distance1*distance1 > 5*5*(dx*dx + dy*dy)){
+                KRATOS_WARNING("Interpolate_sw_to_pfem") << "PFEM node quite far from SW output interface " << std::endl; 
+            }
+
+            if (((dx > tol) & (dy < tol)) | ((dx < tol) & (dy > tol))){
+                if ((dx > tol) & (dy < tol)){
+                    ratio = (it_node->X() - jt_node2->X())/(jt_node1->X() - jt_node2->X());
+                } else if ((dx < tol) & (dy > tol)){
+                    ratio = (it_node->Y() - jt_node2->Y())/(jt_node1->Y() - jt_node2->Y());
+                }
+
+                velocity = ratio*(jt_node1->GetValue(VELOCITY) - jt_node2->GetValue(VELOCITY)) + jt_node2->GetValue(VELOCITY);
```

Comment:
we are not using the `FastGetSolutionStepValue(VELOCITY)` ?

Reply:
The same of comment above